### PR TITLE
Intorduced E2E playwright tests

### PR DESCRIPTION
**Pull request by raya311**
# Pull Request

## Summary

Introduced the playwright to do E2E tests from UI. Currently, they are optional on CI/CD. We need to figure out a few things. 1. Make the static URL work
2. Figure out how to source .env file to the pipleine

## Why

Unit tests only mock the data and do not support integration tests between front and back ends.

## How to Verify

Please follow readme files
1. Set variables on .env file
2. Install node modules (npm install)
3. Install playwright dependencies (npx playwright install) 

Below is the evidence for local test run

image

1. Install dependencies: `npm install`
5. Run the app / relevant command
6. Run lint: `npm run eslint`
7. Run tests: `npm run test`
8. Describe expected behavior

## Checklist (Definition of Done)

- Branch follows team naming pattern (`week<NUM>/<short-description>`)
- PR opened early and kept reviewable in size
- At least one teammate review requested
- CI checks are passing
- Lint passes locally (`npm run eslint`)
- Tests pass locally (`npm run test`)
- New behavior includes starter tests or test plan notes
- Documentation updated if needed (`README`, `docs/`, ADRs)

## Notes for Reviewers

Anything specific you'd like feedback on?
- Edge cases?
- Architecture concerns?
- Naming?
- Test coverage?

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -45,5 +45,40 @@ jobs:
...
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Install Playwright browsers
+        run: npx playwright install --with-deps
+
+      - name: Build application
+        run: npm run build
+
+      - name: Run E2E Tests
```
Comment (hmhawn) [inline .github/workflows/ci.yaml]: This workflow does not provide Google OAuth test credentials or a stable hosted environment, so it is very likely to fail or skip in CI. Since the webServer section in playwright.config.js is commented out, CI also depends on something else already serving the app. Please either:

wire up a real CI-ready environment for Playwright, or
scope this job/documentation clearly as local-only for now.

**Review by hmhawn — CHANGES_REQUESTED** [review]
I’m requesting changes before approval because there are still a few important inconsistencies between the implementation and the documentation.

Main issues to address:

The CI workflow is not fully CI-ready for Playwright yet. The job does not provide Google OAuth test credentials or a stable hosted environment, and with the webServer section commented out, CI depends on something else already serving the app. This should either be fully wired up for CI or documented more clearly as local-only for now.
The documentation currently claims Playwright can automatically start both the backend and frontend, but that is inaccurate in the current code state because the webServer config is commented out. The docs/README should be updated unless that config is re-enabled and verified.
The README test instructions use Unix-style export commands only. Since those won’t work for PowerShell, please add PowerShell equivalents so the instructions are usable across environments.

What looks good:

The PR adds 4 new tests covering primary workflow, integration, and failure-path behavior.
The failure-path coverage is especially relevant to the Week 11 deliverable.
Package scripts and .gitignore updates make sense.

Once the docs are aligned with the actual Playwright setup and the command examples are made cross-platform, this should be in a much better position to approve.
